common/djangoapps/contentserver/middleware.py

```diff
@@ -55,6 +56,25 @@ def process_request(self, request):
             except (ItemNotFoundError, NotFoundError):
                 return HttpResponseNotFound()
 
+            # Set the basics for this request. Make sure that the course key for this
+            # asset has a run, which old-style courses do not.  Otherwise, this will
+            # explode when the key is serialized to be sent to NR.
+            safe_course_key = loc.course_key
+            if safe_course_key.run is None:
+                safe_course_key = safe_course_key.replace(run='only')
```

Comment:
Honestly, I think it's fine as is for now. I don't want to run into another weird corner case because we never have blank runs.